dashboard/test/controllers/courses_controller_test.rb

```diff
@@ -16,12 +16,23 @@ class CoursesControllerTest < ActionController::TestCase
     @pilot_section = create :section, user: @pilot_teacher, unit_group: @pilot_unit_group
     @pilot_student = create(:follower, section: @pilot_section).student_user
 
+    @pilot_facilitator = create :facilitator, pilot_experiment: 'my-pl-experiment'
+    @pilot_pl_unit_group = create :unit_group, pilot_experiment: 'my-pl-experiment', published_state: SharedCourseConstants::PUBLISHED_STATE.pilot, instructor_audience: SharedCourseConstants::INSTRUCTOR_AUDIENCE.facilitator, participant_audience: SharedCourseConstants::PARTICIPANT_AUDIENCE.teacher
+    @pilot_pl_section = create :section, user: @pilot_facilitator, unit_group: @pilot_pl_unit_group
+    @pilot_participant = create :teacher
+    create(:follower, section: @pilot_pl_section, student_user: @pilot_participant)
+
     @unit_group_regular = create :unit_group, name: 'non-plc-course', published_state: SharedCourseConstants::PUBLISHED_STATE.beta
 
     @migrated_unit = create :script, is_migrated: true, published_state: SharedCourseConstants::PUBLISHED_STATE.beta
     @unit_group_migrated = create :unit_group, published_state: SharedCourseConstants::PUBLISHED_STATE.beta
     create :unit_group_unit, unit_group: @unit_group_migrated, script: @migrated_unit, position: 1
 
+    @migrated_pl_unit = create :script, is_migrated: true, published_state: SharedCourseConstants::PUBLISHED_STATE.beta
+    @pl_unit_group_migrated = create :unit_group, published_state: SharedCourseConstants::PUBLISHED_STATE.beta, instructor_audience: SharedCourseConstants::INSTRUCTOR_AUDIENCE.facilitator, participant_audience: SharedCourseConstants::PARTICIPANT_AUDIENCE.teacher
+    create :unit_group_unit, unit_group: @pl_unit_group_migrated, script: @migrated_pl_unit, position: 1
+    @migrated_pl_unit.reload
+
```

Comment:
can you please move all the lines you've added to `setup`, plus lines 14-17, to a new `setup_all` block? it looks like this will shave 30-60 seconds off the time it takes to run this test.

this test was already quite slow to begin with, so if there's anything else you can move into `setup_all` without breaking tests, that would be a nice bonus. as a guideline, if the object is not being modified within individual tests then it is a good candidate to extract to `setup_all`.